pkg/backend/httpstate/client/api.go

```diff
@@ -252,3 +286,36 @@ func pulumiRESTCall(ctx context.Context, diag diag.Sink, cloudAPI, method, path
 
 	return nil
 }
+
+// readBody reads the contents of an http.Response into a byte array, returning an error if one occurred while in the
+// process of doing so. readBody uses the Content-Encoding of the response to pick the correct reader to use.
+func readBody(resp *http.Response) ([]byte, error) {
+	contentEncoding, ok := resp.Header["Content-Encoding"]
+	body := resp.Body
+	defer contract.IgnoreClose(body)
+	if !ok {
+		// no header -> no additional encoding
```

Comment:
Nit: Could you please make this comment a complete sentence? One-two word labels are fine, but this is meaningful/long enough to warrant an actual (and syntactically correct) sentence.

Similarly, the comment below on line 302 should begin with a capital letter. The comment on line 308 should end with a period.

Reply:
done.